framework/packages/abstract-std/src/native_addrs.rs

```diff
@@ -0,0 +1,28 @@
+// TODO: fill bytes with CanonicalAddr of creator
+// Default local-juno used right now(for testing)
+pub const TEST_ABSTRACT_CREATOR: [u8; 20] = [
+    210, 20, 62, 221, 52, 61, 116, 51, 8, 49, 75, 191, 7, 17, 231, 72, 143, 140, 113, 214,
+];
+
+pub const ANS_ADDR: [u8; 32] = [
+    240, 207, 249, 98, 43, 113, 214, 211, 30, 136, 11, 140, 16, 81, 121, 153, 203, 165, 42, 12,
+    186, 176, 183, 241, 120, 71, 168, 99, 151, 45, 246, 17,
+];
+pub const VERSION_CONTROL_ADDR: [u8; 32] = [
```

Comment:
I'm not sure why we need to store all addresses as canonical addresses. Can't we just use instantiate_addr2 using the blob checksum, the creator and the salt. 
In that way, there is only 2 bytes array and N salts to store here ?

Reply:
> I'm not sure why we need to store all addresses as canonical addresses. Can't we just use instantiate_addr2 using the blob checksum, the creator and the salt. In that way, there is only 2 bytes array and N salts to store here ?

We don't need to store it as canonical, but why compute instantiate2 of already known address every time we use any of the native contracts?